paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/FailedEventReplayer.java

```diff
@@ -0,0 +1,88 @@
+package de.zalando.paradox.nakadi.consumer.boot;
+
+import static java.util.Objects.requireNonNull;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Component;
+
+import de.zalando.paradox.nakadi.consumer.boot.components.EventReceiverRegistry;
+import de.zalando.paradox.nakadi.consumer.boot.components.EventTypeConsumer;
+import de.zalando.paradox.nakadi.consumer.boot.components.FailedEventSourceMap;
+import de.zalando.paradox.nakadi.consumer.core.EventHandler;
+import de.zalando.paradox.nakadi.consumer.core.FailedEventSource;
+import de.zalando.paradox.nakadi.consumer.core.domain.EventTypePartition;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+@Component
+public class FailedEventReplayer {
+
+    private final EventReceiverRegistry eventReceiverRegistry;
+
+    private final FailedEventSourceMap failedEventSourceMap;
+
+    private final ReplayHandler replayHandler;
+
+    private static final String EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE = "Event source name is not available.";
+
+    @Autowired
+    public FailedEventReplayer(final EventReceiverRegistry eventReceiverRegistry,
+            final FailedEventSourceMap failedEventSourceMap, final ReplayHandler replayHandler) {
+        this.eventReceiverRegistry = eventReceiverRegistry;
+        this.failedEventSourceMap = failedEventSourceMap;
+        this.replayHandler = replayHandler;
+    }
+
+    public Long getApproximatelyTotalNumberOfFailedEvents(final String eventSourceName) {
+        checkArgument(failedEventSourceMap.getFailedEventSourceMap().containsKey(eventSourceName),
+            EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
+        return failedEventSourceMap.getFailedEventSourceMap().get(eventSourceName).getSize();
+    }
+
+    public Collection<String> getFailedEventSources() {
+        return failedEventSourceMap.getFailedEventSourceMap().keySet();
+    }
+
+    public void replay(final String eventSourceName, final Long numberOfFailedEvents,
+            final boolean breakProcessingOnException) {
+        checkArgument(failedEventSourceMap.getFailedEventSourceMap().containsKey(eventSourceName),
+            EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
+
+        final FailedEventSource<FailedEvent> failedEventSource = failedEventSourceMap.getFailedEventSourceMap().get(
+                eventSourceName);
+        final long approximatelyTotalNumberOfFailedEvents = failedEventSource.getSize();
```

Comment:
`totalNumberOfFailedEvents` is sufficient. Whether it is approximated or not is up to the implementations.